x-pack/platform/plugins/shared/streams/server/lib/rules/esql/register.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AlertInstanceContext, RuleTypeState } from '@kbn/alerting-plugin/server';
+import { alertFieldMap } from '@kbn/alerts-as-data-utils';
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import { ESQL_RULE_TYPE_ID } from '@kbn/rule-data-utils';
+import { PersistenceAlertType } from '@kbn/rule-registry-plugin/server';
+import {
+  STREAMS_FEATURE_ID,
+  STREAMS_RULE_REGISTRATION_CONTEXT,
+} from '../../../../common/constants';
+import { getRuleExecutor } from './executor';
+import { EsqlRuleParams, esqlRuleParams } from './types';
+
+export function esqlRuleType(): PersistenceAlertType<
```

Comment:
is it possible to ship this as a tech preview rule?

Reply:
@elastic/response-ops any advice here?

Reply:
I am not aware of a way to make the rule a tech preview.